### PR TITLE
Forms 10419: clear option bug fix for Date Picker

### PR DESCRIPTION
**Pull request by s1mahanty**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js [hunk 1]
```diff
@@ -184,6 +184,11 @@ describe("Form Runtime with Date Picker", () => {
             cy.get(`#${datePicker7}`).find("input").focus().should("have.value","2/8/2023");
 
         });
+
+        // check clear option
+        cy.get(`#${datePicker7}`).find(".datepicker-calendar-icon").should("be.visible").click().then(() => {
+            cy.get(".dp-clear").click();
```
Comment (devgurjar) [inline ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js:190]: please add another statement where post clicking clear button, the value of date picker should be removed and value should be empty.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 2]
```diff
@@ -989,12 +989,12 @@ if (typeof window.DatePickerWidget === 'undefined') {
...
     #clearDate(view) {
-      this.setValue("");
-      let existingSelectedItem = this['$'
-      + view.toLowerCase()].getElementsByClassName("dp-selected")[0];
+      this.#model.value = "";
+      let existingSelectedItem = this['$' + view.toLowerCase()].getElementsByClassName("dp-selected")[0];
       if (existingSelectedItem) {
         existingSelectedItem.classList.remove("dp-selected");
       }
+      this.#dp.style.display = "none";
```
Comment (devgurjar) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:997]: this is not in parity with AF1, we don't hide date picker dropdown on clicking clear button. But same is happening in HTML5 date picker. 

@rismehta @sakshi-arora1 what should be the expected behaviour.

Comment (iamsudhanshu) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:997]: Are we closing the data picker popup on clicking clear button - if so - we should align with AF1.

Comment (s1mahanty) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:997]: Ok.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 3]
```diff
@@ -1006,8 +1005,10 @@ if (typeof window.DatePickerWidget === 'undefined') {
     }
 
     toString() {
-      return this.selectedYear + "-" + this.#pad2(this.selectedMonth + 1) + "-"
-          + this.#pad2(this.selectedDay);
+      if(this.selectedYear === -1 || this.selectedMonth === -1 || this.selectedDay === -1){
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1008]: if (this.selectedYear === -1 || this.selectedMonth === -1 || this.selectedDay === -1) {
    return "";
}

const formattedDate = `${this.selectedYear}-${this.#pad2(this.selectedMonth + 1)}-${this.#pad2(this.selectedDay)}`;
return formattedDate;

File: ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js [hunk 4]
```diff
@@ -184,6 +184,13 @@ describe("Form Runtime with Date Picker", () => {
             cy.get(`#${datePicker7}`).find("input").focus().should("have.value","2/8/2023");
 
         });
+
+        // check clear option
+        cy.get(`#${datePicker7}`).find(".datepicker-calendar-icon").should("be.visible").click().then(() => {
```
Comment (rismehta) [inline ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js:189]: Can we fix the datepicker-calendar-icon as per BEM,

`cmp-adaptiveform-datepicker__calendar-icon`

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js (hunk 3 above)
Comment (s1mahanty) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1008]: Why do we want a toString function to return a number formattedDate, or did you mean
const formattedDate= this.selectedYear + "-" + this.#pad2(this.selectedMonth + 1) + "-" + this.#pad2(this.selectedDay);

**Pull request merged**